src/main/scala/tech/christopherdavenport/twitterstorm/Resource.scala

```diff
@@ -0,0 +1,47 @@
+package tech.christopherdavenport.twitterstorm
+
+import cats.effect.{Async, Effect}
+import fs2.Stream
+
+abstract class Resource[F[_], A] { self =>
+  def apply[B](f: A => Stream[F, B]): Stream[F, B]
+
+  def serve(implicit F: Async[F]): Stream[F, Nothing] =
+    apply(_ => Stream.eval_(F.async[Nothing](_ => ())))
+
+  def flatMap[B](k: A => Resource[F, B]): Resource[F, B] = {
+    new Resource[F, B] {
+      def apply[C](h: B => Stream[F, C]): Stream[F, C] =
+        self.apply(a => k(a)(h))
+    }
+  }
+
+  def map[B](k: A => B): Resource[F, B] =
+    flatMap(x => Resource.pure(k(x)))
+}
+
+object Resource {
+  def apply[F[_], A](sa: Stream[F, A]): Resource[F, A] =
+    new Resource[F, A] {
+      def apply[B](f: A => Stream[F, B]) = sa.flatMap(f)
+    }
+
+  def eval[F[_], A](fa: F[A]): Resource[F, A] =
+    apply(Stream.eval(fa))
+
+  def pure[F[_], A](a: A): Resource[F, A] =
+    apply(Stream.emit(a))
+
+  def bracket[F[_], A](acquire: F[A])(release: A => F[Unit]): Resource[F, A] =
+    new Resource[F, A] {
+      def apply[B](f: A => Stream[F, B]): Stream[F, B] =
+        Stream.bracket(acquire)(f, release)
```

Comment:
This is why our resource is specialized on `Stream`.  We could probably do something similar with `ApplicativeError`, but this lets us be sure the `release` won't be interrupted.